Makefile

```diff
@@ -54,9 +54,18 @@ docs:
 
 .PHONY: sh-lint
 sh-lint:
-	@(set -e; for f in $$(find . -not \( -path '*/node_modules/*' -prune \) -name '*.sh') ./reach; do \
-		echo linting $$f; \
-		shellcheck --external-sources --source-path SCRIPTDIR $$f ; \
+	@(set -e; for f in \
+	    $$(find . \
+	      -not \( -path '*/node_modules/*' -prune \) \
+	      -not \( -path './hs/app/reach/embed/*' -prune \) \
```

Comment:
I think another option would be to change your use of templates so that the Haskell templates don't use `$` but instead use a character like `@` or `%` and then shellcheck will be able to work with them.